lusidtools/apps/flush_transactions.py

```diff
@@ -0,0 +1,174 @@
+    host : str
+        Host taken from the configuration of the api client. Represents the start the base url involved in the calls
+    input_lst : list
+        A list containing all the transaction information
+    maxCharacterCount : int
+        Maximum number of characters allowed in the URI
+
+    Returns
+    -------
+    batched_data : list of lists
+        Returns a list with each item being a batch (which is itself, a list of transactions)
+
+    """
+
+    # Determine how many characters are fixed to be included in url before the addition of query params
+    base_url = f"{host}/api/transactionportfolios/{scope}/{code}/transactions?"
+    unavoidable_character_count = len(base_url)
+    remaining_count = maxCharacterCount - unavoidable_character_count
+    queryLengthPerTxnID = [len(f"transactionIds={x}&") for x in input_lst]
+
+    # Build the batches
+    batches = []
+    curr_count = 0
+    curr_batch = []
+    for txn_id, str_len in zip(input_lst, queryLengthPerTxnID):
+        if curr_count + str_len >= remaining_count:
+            batches.append(curr_batch)
+            curr_batch = []
+            curr_count = 0
+
+        curr_batch.append(txn_id)
+        curr_count = curr_count + str_len
+
+    # Create remainder final batch if any remainder exists
+    if curr_batch:
+        batches.append(curr_batch)
+
+    return batches
+
+
+def get_ids_from_txns(txns: lusid.VersionedResourceListOfTransaction):
+    """
+        Takes a given input list of transactions and extracts the transaction IDs from them
+
+        Parameters
+        ----------
+        txns : lusid.VersionedResourceListOfTransaction
+            Unedited list of transactions from GetTransactions response
+
+        Returns
+        -------
+        txnIDs : list
+            Returns a list of the transaction IDs for the observed transactions
+    """
+    return [txn.transaction_id for txn in txns.values]
+
+
+def flush(scope, code, fromDate, toDate, api_factory):
+    """
+            Cancels all transactions found in a given date range for a specific portfolio
+
+            Parameters
+            ----------
+            scope : str
+                Scope of the portfolio in question
+            code : str
+                Code of the portfolio in question
+            fromDate : str
+                Date after which transactions will be retrieved
+            toDate : str
+                Date up to which transactions will be retrieved
+        """
+    # Initialise the api
+    secrets_path = str(
+        pathlib.Path(__file__).parent.parent.parent.joinpath("tests", "secrets.json")
+    )
+
+    transaction_portfolios_api = api_factory.build(lusid.api.TransactionPortfoliosApi)
+
+    # Get the Transaction Data
+    print("Getting Transaction Data")
+    transactions = transaction_portfolios_api.get_transactions(
+        scope,
+        code,
+        from_transaction_date=fromDate,
+        to_transaction_date=toDate,
+        limit=5000,
```

Comment:
This limit is just the maximum number of transactions per page as defined by the API so I've done this to set it at its maximum (the default value for this is higher than the max so would break if not hardcoded in)